github/internals/entities.py

```diff
@@ -0,0 +1,757 @@
+    RERUN = "re-run"
+    REBASE = "needs rebase"
+    PRIORITIZED = "prioritized"
+
+
+@unique
+class State(CIEnum):
+    """Enum which represents the status of a commit"""
+    PENDING = "PENDING"
+    FAILURE = "FAILURE"
+    SUCCESS = "SUCCESS"
+    ERROR = "ERROR"
+
+
+class RateLimit(object):
+    """Represents the GitHub API rate limit
+
+    github3py returns a dict with two keys: resources and rate:
+
+    {
+        'rate': {'limit': 5000, 'remaining': 4911, 'reset': 1522063348},
+        'resources': {
+            'core': {'limit': 5000, 'remaining': 4911, 'reset': 1522063348},
+            'graphql': {'limit': 5000, 'remaining': 4927, 'reset': 1522065504},
+            'search': {'limit': 30, 'remaining': 30, 'reset': 1522063078}
+        }
+    }
+
+    So to get the limit we're interested in we need to access the 'resources'
+    dict and get an information.
+    """
+
+    # We don't use 'search' resource, so only 'core' and 'graphql' are valid
+    valid_resources = ["core", "graphql"]
+
+    def __init__(self, limit: int, remaining: int, reset_at: int) -> None:
+        self.limit = limit
+        self.remaining = remaining
+        self.reset_at = reset_at
+
+    @property
+    def available(self) -> bool:
+        return self.remaining >= EPHEMERAL_LIMIT
+
+    def wait(self) -> None:
+        if not self.available:
+            reset_time = datetime.fromtimestamp(self.reset_at)
+            now_time = datetime.now()
+            sleep_time = reset_time - now_time
+
+            sleep(sleep_time.total_seconds())
+
+    @staticmethod
+    def from_dict(data_dict: Dict) -> "RateLimit":
+        """Fabric of RateLimit"""
+        return RateLimit(
+            limit=data_dict["limit"],
+            remaining=data_dict["remaining"],
+            reset_at=data_dict["reset"]
+        )
+
+
+class World(object):
+    """Represents the outside world state"""
+    def __init__(
+        self, graphql_request: Callable, github_api: GitHub,
+        session: Session, repo_owner: Text, repo_name: Text,
+        runner_id: Text, tasks_path: Text, whitelist: List[Text]
+    ) -> None:
+        self.available_resources = AvailableResources()
+        self.graphql_request = graphql_request
+        self.github_api = github_api
+        self.session = session
+        self.repo_owner = repo_owner
+        self.repo_name = repo_name
+        self.runner_id = runner_id
+        self.tasks_path = tasks_path
+        self.whitelist = whitelist
+        self.instance = self
+
+    def get_rate_limit(self, resource: Text=None) -> RateLimit:
+        """Calls GitHub API and returns RateLimit instance"""
+        if resource not in RateLimit.valid_resources:
+            ValueError("Supported resources are graphql and core")
+
+        return RateLimit.from_dict(
+            self.github_api.rate_limit()["resources"][resource]
+        )
+
+    def poll_status(
+        self, pr_number: int, task_name: Text
+    ) -> "Status":
```

Comment:
Shouldn't we use postponed annotation evaluation instead of strings?
https://www.python.org/dev/peps/pep-0563/

Reply:
If you can point what should I change it will be great because in this PEP I see it's about Python 4.0 and Python 3.7

Reply:
ah pff, forget what I said.